Sources/X509/Certificate.swift

```diff
@@ -263,22 +285,41 @@ extension Certificate: DERImplicitlyTaggable {
     @inlinable
     public init(derEncoded rootNode: ASN1Node, withIdentifier identifier: ASN1Identifier) throws {
         self = try DER.sequence(rootNode, identifier: identifier) { nodes in
-            guard let tbsCertificateNode = nodes.next() else {
-                throw ASN1Error.invalidASN1Object(reason: "TBSCertificate missing")
+            guard let tbsCertificateNode = nodes.next(),
+                  let signatureAlgorithmNode = nodes.next(),
+                  let signatureNode = nodes.next()
+            else {
+                throw ASN1Error.invalidASN1Object(reason: "Invalid certificate object, insufficient ASN.1 nodes")
             }
             let tbsCertificate = try TBSCertificate(derEncoded: tbsCertificateNode)
-            let signatureAlgorithm = try AlgorithmIdentifier(derEncoded: &nodes)
-            let signature = try ASN1BitString(derEncoded: &nodes)
-            return try Certificate(tbsCertificate: tbsCertificate, signatureAlgorithm: signatureAlgorithm, signature: signature, tbsCertificateBytes: tbsCertificateNode.encodedBytes)
+            let signatureAlgorithm = try AlgorithmIdentifier(derEncoded: signatureAlgorithmNode)
+            let signature = try ASN1BitString(derEncoded: signatureNode)
+            return try Certificate(
+                tbsCertificate: tbsCertificate,
+                signatureAlgorithm: signatureAlgorithm,
+                signature: signature,
+                tbsCertificateBytes: tbsCertificateNode.encodedBytes,
+                signatureAlgorithmBytes: signatureAlgorithmNode.encodedBytes,
+                signatureBytes: signatureNode.encodedBytes
+            )
         }
     }
 
     @inlinable
     public func serialize(into coder: inout DER.Serializer, withIdentifier identifier: ASN1Identifier) throws {
-        try coder.appendConstructedNode(identifier: identifier) { coder in
-            try coder.serialize(self.tbsCertificate)
-            try coder.serialize(AlgorithmIdentifier(self.signatureAlgorithm))
-            try coder.serialize(ASN1BitString(self.signature))
+        coder.appendConstructedNode(identifier: identifier) { coder in
+            coder.serializeRawBytes(self.tbsCertificateBytes)
+            coder.serializeRawBytes(self.signatureAlgorithmBytes)
+            coder.serializeRawBytes(self.signatureBytes)
         }
     }
 }
+
+extension DER.Serializer {
+    @inlinable
+    static func serialized<Element: DERSerializable>(element: Element) throws -> ArraySlice<UInt8> {
```

Comment:
That is actually a quite common operation. Might be worth to put this into `swift-asn1` in the future